build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -48,6 +54,78 @@ module.exports = function ({template, types: t}) {
     return filename.endsWith('.jss.js');
   }
 
+  function classnameId(name) {
+    return `\$${name}`;
+  }
+
+  /**
+   * @param {Path} reference
+   * @param {string} importedName
+   * @return {?Path}
+   */
+  function findImportDeclaration(reference, importedName) {
+    if (!reference.isIdentifier()) {
+      return null;
+    }
+
+    const binding = reference.scope.getBinding(reference.node.name);
+    if (!binding || binding.kind !== 'module') {
+      return null;
+    }
+
+    const {path} = binding;
+    const {parentPath} = path;
+
+    if (
+      !parentPath.isImportDeclaration() ||
+      !path.isImportSpecifier() ||
+      !t.isIdentifier(path.node.imported, {name: importedName})
+    ) {
+      return null;
+    }
+
+    return parentPath;
+  }
+
+  /**
+   * Replace a MemberExpression (like `a.x` or `useStyles().x`) with its
+   * imported constant (`$x`). It creates the necessary import as well.
+   * @param {Path} importDeclaration
+   * @param {Path} callExpression
+   * @param {Path} memberExpression
+   * @return {boolean}
+   */
+  function replaceMemberExpression(
+    importDeclaration,
+    callExpression,
+    memberExpression
+  ) {
+    const {property} = memberExpression.node;
+    if (!memberExpression.isMemberExpression() || !t.isIdentifier(property)) {
+      return false;
+    }
+
+    const localCalleeName = callExpression.node.callee.name;
+
+    const name = classnameId(property.name);
+    const localName = `${localCalleeName}_${name}`;
+
+    memberExpression.replaceWith(t.identifier(localName));
+
+    // Add classname constant import.
+    const {specifiers} = importDeclaration.node;
+    if (
+      !specifiers.some((specifier) =>
+        t.isIdentifier(specifier.local, {name: localName})
+      )
+    ) {
+      specifiers.push(
+        t.importSpecifier(t.identifier(localName), t.identifier(name))
+      );
+    }
```

Comment:
We can let babel handle this: https://babeljs.io/docs/en/babel-helper-module-imports

Reply:
This has the side effect of creating multiple imports for the same classname. Is that problematic?

Reply:
Nah, Closure will handle it.